inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcOutputFormat.java

```diff
@@ -0,0 +1,497 @@
+        if (batchCount == 0) {
+            return;
+        }
+        checkFlushException();
+
+        for (int i = 0; i <= executionOptions.getMaxRetries(); i++) {
+            try {
+                attemptFlush();
+                batchCount = 0;
+                break;
+            } catch (SQLException e) {
+                LOG.error("JDBC executeBatch error, retry times = {}", i, e);
+                if (i >= executionOptions.getMaxRetries()) {
+                    throw new IOException(e);
+                }
+                try {
+                    if (!connectionProvider.isConnectionValid()) {
+                        updateExecutor(true);
+                    }
+                } catch (Exception exception) {
+                    LOG.error(
+                            "JDBC connection is not valid, and reestablish connection failed.",
+                            exception);
+                    throw new IOException("Reestablish JDBC connection failed", exception);
+                }
+                try {
+                    Thread.sleep(1000 * i);
+                } catch (InterruptedException ex) {
+                    Thread.currentThread().interrupt();
+                    throw new IOException(
+                            "unable to flush; interrupted while doing another attempt", e);
+                }
+            }
+        }
+        // audit, flush metrics data
+        // if (sinkMetricData != null) {
+        // sinkMetricData.flushAuditData();
+        // }
+    }
+
+    protected void attemptFlush() throws SQLException {
+        jdbcStatementExecutor.executeBatch();
+    }
+
+    /** Executes prepared statement and closes all resources of this instance. */
+    @Override
+    public synchronized void close() {
+        if (!closed) {
+            closed = true;
+
+            if (this.scheduledFuture != null) {
+                scheduledFuture.cancel(false);
+                this.scheduler.shutdown();
+            }
+
+            if (batchCount > 0) {
+                try {
+                    flush();
+                } catch (Exception e) {
+                    LOG.warn("Writing records to JDBC failed.", e);
+                    throw new RuntimeException("Writing records to JDBC failed.", e);
+                }
+            }
+
+            // audit, flush metrics data
+            if (sinkMetricData != null) {
+                sinkMetricData.flushAuditData();
+            }
+
+            try {
+                if (jdbcStatementExecutor != null) {
+                    jdbcStatementExecutor.closeStatements();
+                }
+            } catch (SQLException e) {
+                LOG.warn("Close JDBC writer failed.", e);
+            }
+        }
+        connectionProvider.closeConnection();
+        checkFlushException();
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /** Builder for a {@link JdbcOutputFormat}. */
+    public static class Builder {
```

Comment:
this builder is not used ?

Reply:
![image](https://github.com/apache/inlong/assets/58425449/40fe6e7d-1a35-44b6-812f-096b10d2cca9)
It was used in JdbcDynamicTableSink～

Reply:
Already have a class for this which is JdbcOutputFormatBuilder.java, you are actually using this one rather than the inner class in JdbcOutputFormat.java.

Reply:
fix it